packages/asset-uploader/src/schema.ts

```diff
@@ -0,0 +1,41 @@
+import { UploadHandlerPart } from "@remix-run/node";
+import { z } from "zod";
+import { NodeOnDiskFile } from "@remix-run/node";
+import { DEFAULT_UPLPOAD_PATH } from "./constants";
+
+const SingleImageInUpload = z.instanceof(NodeOnDiskFile);
+
+export const ImagesUpload = z.array(SingleImageInUpload);
+
+export type ImagesUpload = z.infer<typeof ImagesUpload>;
+export type SingleImageInUpload = z.infer<typeof SingleImageInUpload>;
+
+export const ImagesUploadedSuccess = z.object({
+  Location: z.string(),
+});
+export type ImagesUploadedSuccess = z.infer<typeof ImagesUploadedSuccess>;
+
+export const s3EnvVariables = z.object({
+  S3_ENDPOINT: z.string(),
+  S3_REGION: z.string(),
+  S3_ACCESS_KEY_ID: z.string(),
+  S3_SECRET_ACCESS_KEY: z.string(),
+  S3_BUCKET: z.string(),
+  S3_ACL: z.string().optional(),
+});
+
+export const fsEnvVariables = z.object({
+  FILE_UPLOAD_PATH: z.string().optional().default(DEFAULT_UPLPOAD_PATH),
+});
+
+export const assetEnvVariables = z.object({
+  MAX_UPLOAD_SIZE: z.number().optional().default(10),
```

Comment:
yeah, should be in bytes

Reply:
That is super confusing if the person has to write it in bytes, I would hate that experrience

Reply:
actually since its coming from env, you won't be able to pass this validation because its going to be a string and zod expects a number, unless z.number does the magical casting somebody hast to cast

Reply:
yeah, I agree regarding writing in bytes, because that would come from env without a nicer js syntax like `10_000`

Reply:
truuue, let me parseInt

Reply:
done!